packages/webpack/css-extract-webpack-plugin/src/CssExtractRspackPlugin.ts

```diff
@@ -275,13 +280,22 @@ class CssExtractRspackPluginImpl {
 
             const cssHotUpdateList = [...asyncChunks, initialChunk].map((
               [chunkName, cssHotUpdatePath],
-            ) => [
-              chunkName!,
-              cssHotUpdatePath!.replace(
+            ) => {
+              // use hash of previous compilation cause CSSHotUpdateRuntimeModule can not get hash immediately
+              const hotUpdatePath = cssHotUpdatePath!.replace(
                 '.css',
                 `${this.hash ? `.${this.hash}` : ''}.css.hot-update.json`,
-              ),
-            ]);
+              );
+              // save all hot update file info
+              this.hotUpdateFiles.set(chunkName!, hotUpdatePath);
+              return [
+                chunkName!,
+                cssHotUpdatePath!.replace(
+                  '.css',
+                  `${this.hash ? `.${this.hash}` : ''}.css.hot-update.json`,
+                ),
```

Comment:
Should this be `hotUpdatePath`?
```suggestion
                hotUpdatePath,
```